FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/workmanager/sync/DeleteDraftsWorker.kt

```diff
@@ -0,0 +1,118 @@
+/*
+ * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
+ * Contributors: DenBond7
+ */
+
+package com.flowcrypt.email.jetpack.workmanager.sync
+
+import android.content.Context
+import androidx.work.Constraints
+import androidx.work.ExistingWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import com.flowcrypt.email.BuildConfig
+import com.flowcrypt.email.api.email.FoldersManager
+import com.flowcrypt.email.api.email.gmail.GmailApiHelper
+import com.flowcrypt.email.api.email.model.LocalFolder
+import com.flowcrypt.email.database.MessageState
+import com.flowcrypt.email.database.entity.AccountEntity
+import com.flowcrypt.email.database.entity.MessageEntity
+import com.sun.mail.imap.IMAPFolder
+import jakarta.mail.Flags
+import jakarta.mail.Folder
+import jakarta.mail.Message
+import jakarta.mail.Store
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+/**
+ * @author Denis Bondarenko
+ *         Date: 9/30/22
+ *         Time: 10:51 AM
+ *         E-mail: DenBond7@gmail.com
+ */
+class DeleteDraftsWorker(context: Context, params: WorkerParameters) :
+  BaseSyncWorker(context, params) {
+  override suspend fun runIMAPAction(accountEntity: AccountEntity, store: Store) {
+    deleteDrafts(accountEntity, store)
+  }
+
+  override suspend fun runAPIAction(accountEntity: AccountEntity) {
+    deleteDrafts(accountEntity)
+  }
+
+  private suspend fun deleteDrafts(account: AccountEntity, store: Store) =
+    withContext(Dispatchers.IO) {
+      deleteDraftsInternal(account) { localFolder, messageEntities ->
+        store.getFolder(localFolder.fullName).use { folder ->
+          val imapFolder = (folder as IMAPFolder).apply { open(Folder.READ_WRITE) }
+          val msgs: List<Message> =
+            imapFolder.getMessagesByUID(messageEntities.map { it.uid }.toLongArray())
+              .filterNotNull()
+          if (msgs.isNotEmpty()) {
+            imapFolder.setFlags(msgs.toTypedArray(), Flags(Flags.Flag.DELETED), true)
+          }
+        }
+
+        roomDatabase.msgDao().deleteSuspend(messageEntities)
+      }
+    }
+
+  private suspend fun deleteDrafts(account: AccountEntity) = withContext(Dispatchers.IO) {
+    deleteDraftsInternal(account) { localFolder, messageEntities ->
+      executeGMailAPICall(applicationContext) {
+        GmailApiHelper.deleteDrafts(
+          context = applicationContext,
+          accountEntity = account,
+          ids = messageEntities.mapNotNull { it.draftId }
```

Comment:
I see it used here, where does it come from?

Reply:
From the local database. We marked a draft to be deleted. 

https://github.com/FlowCrypt/flowcrypt-android/blob/eca36ad7826c741b87a7ea5f5c7c11a81db430c4/FlowCrypt/src/main/java/com/flowcrypt/email/database/MessageState.kt#L48

The similar to logic as we do deleting regular messages.